school_center.py

```diff
@@ -135,43 +133,118 @@ def is_allocated(scode1: str, scode2:str) -> bool:
         return allocations[scode1].get(scode2) != None
     else:
         return False
+    # check if the file exists
+def validate_school_file(file_path: str) -> bool:
+    '''
+    validating the format of the school input file
+    '''
+    if not os.path.exists(file_path):
+        print(f"School file '{file_path}' not found.")
+        return False
+    
+    # open the file and read the header
+    with open(file_path, 'r', newline='', encoding='utf-8') as file:
+        header = file.readline().strip().split('\t')
+        required_columns = ['scode', 'count', 'name-address', 'lat', 'long']
+        if header != required_columns:
```

Comment:
this seems to mandate columns be in this exact order. if so this is too rigid. just check if the required columns are present. even if extra columns are present, it is okay to continue